### DISCUSSION TITLE
source for 3rd party wedo/spike-prime compatible sensors?

### DISCUSSION THREAD

**Discussion by m10d** [pybricks/support#1671] [Q&A]
I bought a PyBricks license to switch to their blocks language for my kids.  Two days in, it's MUCH better than legos.    Appreciate the team's efforts - and pretty crazy how far past lego it is in "quality of life", especially given the resources LEGO group has :(

I'm looking for a few more sensors.  Paying $30-40 each for OEM seems pretty steep.  I do have rPi's and the build hat - but for this age I strongly prefer things buttoned-up inside system or technic compatible housings.    Is anyone aware of any sources for per-integrated wedo-connector compatible sensors? I do have a printer, but would prefer to buy pre-made sensors if they exist; thankfully does appear it'd be easy to buy  [VL6180X distance sensor boards](https://www.adafruit.com/product/3316) and run [MyOwnBricks](https://github.com/ysard/MyOwnBricks) as a backup.  The only vendor I could find is MindSensors, which has a limited set, also pretty pricey (albeit higher functionality than Lego's).    There are a few on aliexpress, but I can't find any reviews or (as is common) any way to determine exactly what brand/quality I'd be buying.

I'd like to buy/make a handful each of distance (10-30ish cm), force, color (and open to others that would be fun to program against) - so the cost adds up quite fast.

Thanks!

**Comment by laurensvalk** [comment]
If you don't have it already, I would like to recommend getting the [color and distance](https://brickset.com/sets/88007-1). It's just $15 on the LEGO shop and does a lot for the price.

Besides color and rudimentary distance, you could get creative and turn it into a button too by detecting the reflectivity of your skin. And even a force sensor if you add a lever with a rubber band and use the reflection value as a measure for how close it is.

Additionally, any of the motors you have are great as inputs too, since you can measure their position and use them as an input dial. Or again as a force sensor it you combine the dial with some feedback power.

**Reply by m10d** [reply]
I can find that sensor via search in the GB locale lego store, but it 404's in en-us? Is there a way to get lego to ship to the US from abroad for a reasonable fee?  (Also, whats the distance-senaing range on that guy?)

**Reply by dlech** [reply]
There are plenty to be had in the US on [BrickLink](https://www.bricklink.com/v2/catalog/catalogitem.page?P=bb0891c01&name=Electric%20Sensor,%20Color%20and%20Distance%20-%20Boost&category=%5BElectric%5D#T=S&C=1&O={%22color%22:1,%22ss%22:%22US%22,%22loc%22:%22US%22,%22iconly%22:0}).

**Reply by m10d** [reply]
> There are plenty to be had in the US on [BrickLink](https://www.bricklink.com/v2/catalog/catalogitem.page?P=bb0891c01&name=Electric%20Sensor,%20Color%20and%20Distance%20-%20Boost&category=%5BElectric%5D#T=S&C=1&O=%7B%22color%22:1,%22ss%22:%22US%22,%22loc%22:%22US%22,%22iconly%22:0%7D).

Thanks for the direct link - those do look reasonable, i must have botched my search as I saw much higher prices when I glanced earlier.

Are you familiar with the effective distance of the sensing on these?

**Reply by laurensvalk** [reply]
Oh, I thought it was $13 on the US LEGO site the other day. Sorry about that.

The distance sensing is just approximate IR light reflection in just 10 increments. Still fun for quite a few use cases. And slightly more useful in practice than the Ultrasonic Sensor (IMO) which just about only works with a nearby flat solid object exactly perpendicular it.

**Comment by BertLindeman** [comment]
Completely separate from pybricks, this is on my personal account. (Laurens, remove if out-of-order here)
On lego.com I cold not find it.

I looked on bricklink.com and brickowl.com (I use them both sometimes to buy parts) 
On both platforms I see one seller each, both located in The United Kingdom.
Both have 2 new sensors. 

You might also look on [rebrickable.com](https://rebrickable.com/parts/26912/sensor-color-and-distance-powered-up-2-x-4-x-2/#buy_parts)

**Reply by m10d** [reply]
> Completely separate from pybricks, this is on my personal account. (Laurens, remove if out-of-order here) On lego.com I cold not find it.
> 
> I looked on bricklink.com and brickowl.com (I use them both sometimes to buy parts) On both platforms I see one seller each, both located in The United Kingdom. Both have 2 new sensors.
> 
> You might also look on [rebrickable.com](https://rebrickable.com/parts/26912/sensor-color-and-distance-powered-up-2-x-4-x-2/#buy_parts)

Thanks; with dlech's link above I see the old wedo color&distance sensors are probably a good bet; ordering some.

Still looking for a contact/force sensor that makes sense.  I might end up with LMS-esp32 and multiplex some cheap switches that way.  We'll see; for the kids' age it would be great to have the more durable OEM-style.
